object-model/copy-semantics.md

```diff
@@ -0,0 +1,151 @@
+# C++ object model: copy semantics
+_Skeleton instructions are typeset in italic text._
+
+## Overview
+
+_Provides a short natural language abstract of the module’s contents._
+_Specifies the different levels of teaching._
+
+<table>
+  <thead>
+    <th>Level</th>
+    <th>Objectives</th>
+  </thead>
+  <tr>
+    <td>Foundational knowledge</td>
+    <td>understanding how and when are copies made</td>
+  </tr>
+  <tr>
+    <td>Advanced</td>
+    <td>implementing user-defined copy operations</td>
+  </tr>
+  <tr>
+    <td>Further studies</td>
+    <td>special cases: copy elision</td>
+  </tr>
+</table>
+
+## Motivation
+
+_Why is this important?_
+_Why do we want to learn/teach this topic?_
+
+Copy semantics allows the user to determine how objects of a class get replicated and interact on a value level.
+
+## Topic introduction
+
+_Very brief introduction to the topic._
+
+Explains when and how objects are copied.
+
+## Foundational knowledge: How and when are copies made
+
+### Background/Required Knowledge
+
+A student:
+* knows what a type is?             [[C++ object model: types]][1]
+* knows what an object is?          [[C++ object model: objects]][2], [[C++ object model: constant objects]][3]
+* knows what class invariants are?
+
+It helps when a student:
+* knows move semantics              [[C++ object model: move semantics]][4]
+* knows special member functions    [[C++ object model: special member functions]][5]
+
+### Student outcomes
+
+_A list of things “a student should be able to” after the curriculum._
+_The next word should be an action word and testable in an exam._
+_Max 5 items._
+
+A student should be able to:
+
+1. explain what copy semantics accomplish
+  * establishing “equivalent” object state in another object
+2. explain difference between deep and shallow copy
+3. explain where copies are made
+
+### Caveats
+
+_This section mentions subtle points to understand, like anything resulting in
+implementation-defined, unspecified, or undefined behavior._
+
+* Compiler-provided (shallow) copies may result in ownership problems (e.g. `char*`).
+
+### Points to cover
+
+_This section lists important details for each point._
+
+* Principle of copying
+  * Copying of trivially copyable types
+  * Copying of non-trivial class types
+  * Copying an object does not change the original
+* Copy Flavors
+  * Shallow copy
+  * No copy
+  * Deep copy
+* Practical applications
+  * Unique_ptr  (no copy)
+  * Strings (deep copy) 
+
+
+## Advanced: Implementing user-defined copy operations
+
+### Background/Required Knowledge
+
+A student:
+* knows special member functions  [C++ object model: special member functions
```

Comment:
"A student knows" is an objective, not an outcome. Perhaps "A student can identify the special member functions".
Also, your link isn't complete.

Reply:
A quick reminder on the distinction between objectives and outcomes:

An *objective* is some goal that we aim for. An *outcome* is something that we use to measure that the objective has been satisfied. While objectives can be intangible (and as @jcvw puts it, fluffy), outcomes must be a measurable metric.

Reply:
Yes, but should we formulated required knowledge as outcomes? Up until now, I thought this wording distinction was only important for the student goals (i.e., the "A student should be able to" section).

Regarding the "broken link":
I added all required links but most of the docs are currently missing. Should I just add empty files as place holders so the links work or is it ok to have them "broken" until we add the corresponding document?

Reply:
@cjdb I reworded the Background section to a "outcome" oriented style. Do you have further remarks?